docs/sources/operations/request-validation-rate-limits.md

```diff
@@ -0,0 +1,226 @@
+|-------------------------|-------------------------|
+| Enforced by             | `ingester`              |
+| Outcome                 | Request rejected        |
+| Retryable               | Yes                     |
+| Sample discarded        | No                      |
+| Configurable per tenant | Yes                     |
+| HTTP status code        | `429 Too Many Requests` |
+
+## Validation Errors
+
+Validation errors occur when a request violates a validation rule defined by Loki.
+
+### `line_too_long`
+
+This error occurs when a log line exceeds the maximum allowable length in bytes.
+
+This value can be modified globally in the [`limits_config`](https://grafana.com/docs/loki/latest/configuration/#limits_config) block, or on a per-tenant basis in the [runtime overrides](https://grafana.com/docs/loki/latest/configuration/#runtime-configuration-file) file. To increase the maximum line size, adjust `max_line_size`.  Alternatively, logs can be truncated if they exceed the maximum length in bytes with `max_line_size_truncate`.
+
+| Property                | Value            |
+|-------------------------|------------------|
+| Enforced by             | `distributor`    |
+| Retryable               | **No**           |
+| Sample discarded        | **Yes**          |
+| Configurable per tenant | Yes              |
+
+### `invalid_labels`
+
+This error occurs when one or more labels in the submitted streams fail validation.
+
+Loki uses the [same validation rules as Prometheus](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) for validating labels.
+
+> Label names may contain ASCII letters, numbers, as well as underscores. They must match the regex `[a-zA-Z_][a-zA-Z0-9_]*`. Label names beginning with __ are reserved for internal use.
+
+| Property                | Value         |
+|-------------------------|---------------|
+| Enforced by             | `distributor` |
+| Retryable               | **No**        |
+| Sample discarded        | **Yes**       |
+| Configurable per tenant | No            |
+
+## `missing_labels`
+
+This validation error is returned when a stream is submitted without any labels.
+
+| Property                | Value         |
+|-------------------------|---------------|
+| Enforced by             | `distributor` |
+| Retryable               | **No**        |
+| Sample discarded        | **Yes**       |
+| Configurable per tenant | No            |
+
+## `too_far_behind` and `out_of_order`
+
+The `too_far_behind` and `out_of_order` reasons are identical, except that `out_of_order` can only occur if the `unordered_writes` config option is set to `false`. The default value since Loki v2.4 is `true`, so `too_far_behind` is more common.
+
+This validation error is returned when a stream is submitted out of order. More details can be found [here](https://grafana.com/docs/loki/latest/configuration/#accept-out-of-order-writes) about Loki's ordering constraints.
```

Comment:
does the http response indicate which stream has the ooo samples?

Reply:
We don't kill the HTTP request on this error, so there's no response containing the streams.